mlir/lib/Dialect/Polynomial/IR/PolynomialAttributes.cpp

```diff
@@ -83,7 +83,7 @@ parseMonomial(AsmParser &parser, Monomial &monomial, llvm::StringRef &variable,
 
     // If there's a **, then the integer exponent is required.
     APInt parsedExponent(apintBitWidth, 0);
-    if (failed(parser.parseInteger(parsedExponent))) {
+    if (failed(parser.parseInteger(parsedExponent, apintBitWidth, true))) {
```

Comment:
this probably ought be signless (cc @j2kun) can you make apSignless a constexpr variable like https://github.com/llvm/llvm-project/blob/c3276a96d909233b836e839a23a7ad510fae407a/mlir/include/mlir/Dialect/Polynomial/IR/Polynomial.h#L30